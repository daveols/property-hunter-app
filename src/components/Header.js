import React from 'react';
import {Appbar} from 'react-native-paper';

const Header = ({scene, previous, navigation}) => {
  const {options} = scene.descriptor;

  const title =
    options.headerTitle !== undefined
      ? options.headerTitle
      : options.title !== undefined
      ? options.title
      : scene.route.name;

  const rightIcons = options.rightIcons ? options.rightIcons : [];

  return (
    <Appbar.Header>
      {previous && <Appbar.BackAction onPress={navigation.goBack} />}
      <Appbar.Content title={title} />
      {rightIcons.map(({navTo, ...props}, i) => (
        <Appbar.Action
          key={`action-${i}`}
          {...props}
          onPress={() => navigation.navigate(navTo)}
        />
      ))}
    </Appbar.Header>
  );
};

export default Header;
