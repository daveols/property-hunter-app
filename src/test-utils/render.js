import React from 'react';
import {render} from '@testing-library/react-native';

import {DomainAuthProvider} from 'contexts/DomainAuthContext';

export default (ui, {...options} = {}) => {
  const Providers = ({children}) => (
    <DomainAuthProvider>{children}</DomainAuthProvider>
  );

  return {...render(ui, {wrapper: Providers, ...options})};
};
