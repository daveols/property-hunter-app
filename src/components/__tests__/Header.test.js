import React from 'react';

import render from 'test-utils/render';
import {fireEvent} from '@testing-library/react-native';

import Header from 'components/Header';

const baseProps = {
  scene: {
    descriptor: {
      options: {},
    },
    route: {
      name: 'Awesome App',
    },
  },
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
};

describe('Header', () => {
  beforeEach(() => jest.resetAllMocks());

  test('matches snapshot', () => {
    const {container, getAllByRole} = render(<Header {...baseProps} />);

    expect(getAllByRole('header')[0]).toHaveTextContent('Awesome App');
    expect(container).toMatchSnapshot();
  });

  test('back button shows when previous route exists', () => {
    const props = {
      ...baseProps,
      previous: {},
    };
    const {queryByLabelText} = render(<Header {...props} />);

    expect(queryByLabelText('Back')).toMatchSnapshot();
  });

  test('renders right icons and fires correct navigation event when pressed', () => {
    const rightIcons = [
      {
        icon: 'magnify',
        navTo: 'TestScreen',
      },
    ];
    const props = {
      ...baseProps,
      scene: {
        ...baseProps.scene,
        descriptor: {
          options: {
            rightIcons,
          },
        },
      },
    };
    const {getAllByRole} = render(<Header {...props} />);

    const icon = getAllByRole('button')[0];
    fireEvent.press(icon);

    expect(baseProps.navigation.navigate).toBeCalledWith('TestScreen');
  });
});
