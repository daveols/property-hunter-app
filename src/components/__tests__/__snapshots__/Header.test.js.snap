// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header back button shows when previous route exists 1`] = `
<TouchableHighlight
  accessibilityComponentType="button"
  accessibilityLabel="Back"
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessibilityTraits="button"
  activeOpacity={0.85}
  centered={true}
  delayPressOut={100}
  disabled={false}
  hitSlop={
    Object {
      "bottom": 6,
      "left": 6,
      "right": 6,
      "top": 6,
    }
  }
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 6,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 18,
          "height": 36,
          "width": 36,
        },
        undefined,
        undefined,
      ],
    ]
  }
  underlayColor="rgba(255, 255, 255, 0.16)"
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "height": 24,
            "width": 24,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
                "width": 24,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../react-native-paper/lib/commonjs/assets/back-chevron.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 21,
                  "resizeMode": "contain",
                  "width": 21,
                },
                Object {
                  "tintColor": "#ffffff",
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</TouchableHighlight>
`;

exports[`Header matches snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <SafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
      <TouchableWithoutFeedback>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "paddingHorizontal": 12,
              },
              Array [
                false,
                Object {
                  "alignItems": "center",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            accessibilityRole="header"
            accessibilityTraits="header"
            accessible={true}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "fontSize": 17,
                  },
                  undefined,
                ],
              ]
            }
          >
            Awesome App
          </Text>
        </View>
      </TouchableWithoutFeedback>
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
    </View>
  </SafeAreaView>
</View>
`;
